app/jobs/cleanup_expired_stored_mails_job.rb

```diff
@@ -0,0 +1,29 @@
+class CleanupExpiredStoredMailsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    # Really destroy already moderated emails older than two weeks
+    deleted_mails = StoredMail.only_deleted.where('created_at <= ?', 4.weeks.ago)
+    deleted_mails_count = deleted_mails.count
+    deleted_mails.each(&:really_destroy!) if deleted_mails_count.positive?
+
+    # Delete all expired (ignored) emails after the 3-day threshold imposed by Mailgun
+    expired_mails = StoredMail.where('created_at <= ?', 4.days.ago)
+    expired_mails_count = expired_mails.count
+    expired_mails.each(&:destroy) if expired_mails.any?
+
+    inform_slack(deleted_mails_count, expired_mails_count)
+  end
+
+  def inform_slack(deleted_count, expired_count)
+    if deleted_count.positive? && expired_count.positive?
```

Comment:
If you early return this, it becomes easier to read the way smaller `else` branch.